proposals/public-suffix.md

```diff
@@ -0,0 +1,555 @@
+will no longer be duplicating work already done by the host browser.
+
+### Features of the PSL
+
+In order to set out the use cases fully, it is first necessary to describe certain
+relevant features of the PSL.
+
+#### 1. ICANN vs Private
+
+The PSL it is divided into two sections: ICANN suffixes and Private (i.e. non-ICANN)
+suffixes.
+
+##### Examples
+
+| ICANN Suffixes                | Private Suffixes (Non-ICANN)  |
+|-------------------------------|------------------------------ |
+| com                           | blogspot.com                  |
+|                               | members.linode.com            |
+|                               | simplesite.com                |
+|                               |                               |
+| co.uk                         | blogspot.co.uk                |
+|                               | wellbeingzone.co.uk           |
+|                               | vm.bytemark.co.uk             |
+
+The suffixes in the PSL represent boundaries between registrar-type organizations
+and their clients. The "owner" of the suffix is one organization, and the "owner"
+of a domain label immediately preceeding the suffix is a different organization.
+
+In the case of suffixes in the ICANN section, the "owner" of a suffix is ICANN /
+an affiliated registrar. In the case of suffixes in the Private section, the
+"owner" of a suffix is a private organization offering a service for clients
+to take ownership of subdomains underneath its own ICANN-suffixed domain.
+
+#### 2. Multiple Suffixes per Domain
+
+If a domain name ends with a suffix listed in the Private section of the PSL,
+then it must also end with a shorter suffix in the ICANN section. The question
+of which one should be taken to be the domain's public suffix depends on the
+specific use case.
+
+##### Example
+
+| Domain name         | Private suffix  | ICANN suffix |
+|:-------------------:|:---------------:|:------------:|
+| foo.bar.wixsite.com | wixsite.com     | com          |
+
+#### 3. Known vs Unknown
+
+If a domain name does not match any suffix in the PSL, it is considered to have
+an unknown suffix. Depending on the use case, the public suffix could be taken
+by default to be the last domain label of the domain name, or alternatively
+the domain name could be considered invalid.
+
+**Note:** it may be more performant to allow unknown suffixes and assume a single-label
+suffix by default, because it allows the following optimisation of the lookup algorithm:
+all single-label suffixes in the PSL can be excluded from the lookup, since they do not
+need to be matched specifically.
+
+#### 4. IDN
+
+The PSL uses Unicode, and contains International Domain Name (IDN) suffixes. Depending
+on the use case, when returning the *registrable domain / eTLD+1* for a domain name,
+either Unicode or Punycode may be preferred.
+
+**Note:** the [PSL algorithm](https://github.com/publicsuffix/list/wiki/Format#formal-algorithm)
```

Comment:
I think this might be a made up requirement. (That wiki page really needs some work.) The algorithm is pretty trivial and just needs to compare labels for equality so any encoding should be fine.